lib/features/context-pad/ContextPad.js

```diff
@@ -422,18 +405,35 @@ ContextPad.prototype.getPad = function(target) {
     event.stopPropagation();
   });
 
-  var activeRootElement = this._canvas.getRootElement();
-
-  this._overlayId = overlays.add(activeRootElement, 'context-pad', overlaysConfig);
-
-  var pad = overlays.get(this._overlayId);
+  this._container.appendChild(html);
 
   this._eventBus.fire('contextPad.create', {
     target: target,
-    pad: pad
+    pad: html
   });
 
-  return pad;
+  return html;
+};
+
+/**
+ * @param {ContextPadTarget} target
+ *
+ * @return { { html: HTMLElement } }
+ */
+ContextPad.prototype.getPad = function(target) {
+  console.warn(new Error(`ContextPad.getPad is deprecated and will be removed in future library versions;
```

Comment:
We _could_ simply link the PR as a reference and update it with proper deprecation notes; this way we can change it based on user feedback, and don't have a fixed debatable practice (access internals) in the logs.

If folks use the `getPad` for something relevant, let's allow them to raise their concerns.

```suggestion
  console.warn(new Error('ContextPad#getPad is deprecated and will be removed in future library versions, cf. https://github.com/bpmn-io/diagram-js/pull/888'));
```